File: app/src/main/java/com/ebsoft/watchlist/data/control/db/DBManager.java
package com.ebsoft.watchlist.data.control.db;

import android.arch.lifecycle.LiveData;

import com.ebsoft.watchlist.data.model.db.Stock;
import com.ebsoft.watchlist.data.model.db.Watchlist;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by endre on 09/09/18.
 */

public interface DBManager {

    LiveData<List<Watchlist>> loadWatchlists();

    LiveData<List<Stock>> loadStocks(Watchlist watchlist);

    Observable<Boolean> deleteWatchlist(Watchlist watchlist);

    Observable<Boolean> saveWatchlist(Watchlist watchlist);

    Observable<List<Stock>> queryStock(String symbol);

    Observable<Boolean> deleteStock(Stock stock);

    Observable<Boolean> insertStock(Stock stock);

    Observable<Boolean> updateStock(Stock stock);
}

File: app/src/main/java/com/ebsoft/watchlist/data/control/db/DBManager.kt
package com.ebsoft.watchlist.data.control.db

import android.arch.lifecycle.LiveData

import com.ebsoft.watchlist.data.model.db.Stock
import com.ebsoft.watchlist.data.model.db.Watchlist

import io.reactivex.Observable

/**
 * Created by endre on 09/09/18.
 */

interface DBManager {

    fun loadWatchlists(): LiveData<List<Watchlist>>

    fun loadStocks(watchlist: Watchlist): LiveData<List<Stock>>

    fun deleteWatchlist(watchlist: Watchlist): Observable<Boolean>

    fun saveWatchlist(watchlist: Watchlist): Observable<Boolean>

    fun queryStock(symbol: String): Observable<List<Stock>>

    fun deleteStock(stock: Stock): Observable<Boolean>

    fun insertStock(stock: Stock): Observable<Boolean>

    fun updateStock(stock: Stock): Observable<Boolean>
}
